one_fm/api/doc_methods/shift_request.py

```diff
@@ -331,16 +377,22 @@ def fill_to_date(doc, method):
         doc.to_date = doc.from_date
 
 def validate_from_date(doc, method):
-    if not doc.assign_day_off:
+    if doc.purpose != 'Assign Day Off' and not (frappe.session.user == get_employee_user_id(frappe.get_doc("ONEFM General Setting").get("attendance_manager"))):
         if getdate(today()) > getdate(doc.from_date):
-            frappe.throw('From Date cannot be before today.')
+            frappe.throw(
+                _("Please note that Shift Requests cannot be created for a past date."),
+                title=_("Invalid Start Date"),
+		    )
 
 @frappe.whitelist()
 def update_request(shift_request, from_date, to_date):
     from_date = getdate(from_date)
     to_date = getdate(to_date)
     if getdate(today()) > from_date:
-        frappe.throw('From Date cannot be before today.')
+        frappe.throw(
+                _("Please note that Shift Requests cannot be updated to a past date."),

```

Comment:
having this method block in more than one place, so create a method and call it.

Reply:
It's a different error.